testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -83,25 +143,42 @@ async fn delete_k8s_collection<T: Clone + DeserializeOwned + Meta>(
     {
         either::Left(list) => {
             let names: Vec<_> = list.iter().map(Meta::name).collect();
-            println!("Deleting collection of {}: {:?}", name, names);
+            info!("Deleting collection of {}: {:?}", name, names);
         }
         either::Right(status) => {
-            println!("Deleted collection of {}: status={:?}", name, status);
+            info!("Deleted collection of {}: status={:?}", name, status);
         }
     }
 
     Ok(())
 }
 
-pub(crate) async fn delete_k8s_cluster() -> Result<()> {
+pub(crate) async fn delete_k8s_cluster(kube_namespace: String) -> Result<()> {
     let client: K8sClient = create_k8s_client().await;
-    let deployments: Api<Deployment> = Api::namespaced(client.clone(), "default");
-    let stateful_sets: Api<StatefulSet> = Api::namespaced(client, "default");
 
-    // delete all deployments and statefulsets
-    // cross this with all the compute resources created by aptos-node helm chart
-    delete_k8s_collection(deployments, "deployments").await?;
-    delete_k8s_collection(stateful_sets, "stateful_sets").await?;
+    // if operating on the default namespace,
+    match kube_namespace.as_str() {
+        "default" => {
+            let deployments: Api<Deployment> = Api::namespaced(client.clone(), "default");
+            let stateful_sets: Api<StatefulSet> = Api::namespaced(client, "default");
+
+            // delete all deployments and statefulsets
+            // cross this with all the compute resources created by aptos-node helm chart
+            delete_k8s_collection(deployments, "deployments").await?;
+            delete_k8s_collection(stateful_sets, "stateful_sets").await?;
+        }
+        s if s.starts_with("forge") => {
```

Comment:
it seems we only allow deleting namespace which start with `forge`, but we allow creating/running forge in namespaces of any name.
I think we should actually have some namespace name validation logic (e.g. needs to start with `forge-` (notice the `-`) and enforce this name constraint both for the test and cleanup command.